ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -614,4 +632,17 @@ private[kvutils] object TransactionCommitter {
   def getContractState(commitContext: CommitContext, key: DamlStateKey): DamlContractState =
     commitContext.get(key).getOrElse(throw Err.MissingInputState(key)).getContractState
 
+  @SuppressWarnings(Array("org.wartremover.warts.Option2Iterable"))
+  private def transactionMinimumRecordTime(
+      maybeDeduplicateUntil: Option[Instant],
+      ledgerTime: Instant,
+      maxSkew: Duration,
+      submissionTime: Instant): Instant =
+    List(
+      maybeDeduplicateUntil,
+      Some(ledgerTime.minus(maxSkew)),
+      Some(submissionTime.minus(maxSkew))).flatten.max
+
+  private def transactionMaximumRecordTime(ledgerTime: Instant, minSkew: Duration): Instant =
+    ledgerTime.plus(minSkew)
```

Comment:
You want to test also the submission time skew here.

Reply:
Could you expand on that or point me to the relevant definitions? Not really sure about the correct check here.

Reply:
You want to make sure that record time - min_skew <= submission time <= record time. So you need to check record time <= submission time + min_skew, just like for ledger time.

Reply:
Thanks, addressed.